[Workers] (WIP) Assets (#16610)
cloudflare/cloudflare-docs: src/content/docs/workers/static-assets/index.mdx

---
pcx_content_type: navigation
title: Static Assets
sidebar:
  order: 6
  group:
    badge: Beta
head: []
description: Create full-stack applications deployed to Cloudflare Workers.
---

import {
	Badge,
	Description,
	InlineBadge,
	DirectoryListing,
	Render,
	TabItem,
	Tabs,
	Feature,
} from "~/components";

<Description>
	Create fast, scalable frontend applications deployed to Cloudflare Workers.
</Description>

---

### What you can build

<Feature header="Static Websites" href="/workers/frameworks/get-started/#deploying-a-static-site" cta="Deploy a static site">

Deploy static sites like blogs, portfolios, or documentation pages. Serve HTML, CSS, JavaScript, and media files directly from Cloudflare's network for low-latency access worldwide.

</Feature>

> <Feature header="Full-stack Applications" href="/workers/frameworks/get-started/#deploying-a-dynamic-site" cta="Deploy a full-stack application">

Author: updating format to match workers sites: https://developers.cloudflare.com/workers/configuration/sites/

:::note

If you're looking for a framework not on this list, it may be supported in [Cloudflare Pages](/pages/). For example, popular frameworks such as [Next.js](/pages/framework-guides/nextjs/) and [Vue](/pages/framework-guides/deploy-a-vue-site/) are supported on Pages. Visit [Pages Frameworks guides](/pages/framework-guides/) for a full list.

Have a framework you would like to see supported on Workers or a feature request? Tell us about it on [Cloudflare's Developer Discord](https://discord.gg/dqgZUwcD)!
:::

### Pricing

Requests to static assets are free and unlimited. Refer to [pricing](/workers/platform/pricing/#example-2) for an example.

### Limitations

The following limitations apply for Workers with assets:

- [Smart Placement](/workers/configuration/smart-placement/) cannot be enabled for projects with Assets. We are actively working on removing this limitation.
- Assets cannot be uploaded to [Workers for Platforms](/cloudflare-for-platforms/workers-for-platforms/) [user workers](/cloudflare-for-platforms/workers-for-platforms/reference/how-workers-for-platforms-works/). We are working on removing this limitation during the beta period.

> - [RPC](/workers/runtime-apis/bindings/service-bindings/rpc/) communication between service bindings are not supported in `wrangler dev` for Workers with assets. We are working on removing this limitation during the beta period.

Reviewer: ```suggestion
- In local development, you cannot make [Service Binding RPC calls](/workers/runtime-apis/bindings/service-bindings/rpc/) to a Worker with static assets. This is a temporary limitation — we are working to remove it.
```

> - Workers with assets cannot run on a [route or domain](/workers/configuration/routing/) with a path. Eg. `example.com` is an acceptable route but `example.com/foo` is not and will throw an error.

Reviewer: ```suggestion
- You cannot add a [route or domain](/workers/configuration/routing/) that has a path to a Worker that has static assets. Eg. `example.com` is an acceptable route but `example.com/foo` is not and will throw an error.
```

We need to be _specific_ about what type of error. A runtime error? An error when you try to add the route? etc.
